core/icon.js

```diff
@@ -26,6 +26,7 @@
 
 goog.provide('Blockly.Icon');
 
+goog.require('Blockly.BlockSvg');
```

Comment:
Same here.

Reply:
Q about typing: So if the icon is required by the block svg, the icon can reference block svg members?

Reply:
Not a typing question per se, but a closure compiler question, and the answer is a little complex: 
You've identified there's a circular dependency there, ``Block`` requires ``Icon``, and ``Icon`` requires ``Block``.

We know that we will never instantiate an icon in an environment that doesn't have ``Blockly.BlockSvg`` included, so you can choose to just leave it out, and we do that in much of Blockly.
However, the same reason why you thought you needed it, it looks a little odd using and expecting it in ``Icon``, but not requiring it.

The closure compiler does have a term for this: forward declaration. 
Here, you can use: ``goog.forwardDeclare('Blockly.BlockSvg')`` basically stating that you need it for this "module" but it will be declared in the future.